Task 2/2.1/2.1.2/Figures.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace _2._1._2
+{
+    public abstract class Figure
+    {
+        public string name { protected set;  get; }
```

Comment:
```suggestion
        public string Name { get; protected set; }
```

There's no need to store figure name here if every figure still has to use use it themselves. You could use it here in ToString and the call base.ToString in children